nixos/modules/profiles/minimal.nix

```diff
@@ -10,7 +10,17 @@ with lib;
 
   documentation.enable = mkDefault false;
 
+  documentation.doc.enable = mkDefault false;
+
+  documentation.info.enable = mkDefault false;
+
+  documentation.man.enable = mkDefault false;
+
   documentation.nixos.enable = mkDefault false;
 
   programs.command-not-found.enable = mkDefault false;
+
+  services.logrotate.enable = mkDefault false;
```

Comment:
That could be really dangerous and fill up your disk when logs in containers get no longer rotated. Also many people don't even know that the container uses this profile.

Reply:
This module is used only in tests and in minimal ISO image. Should not affect working to containers. And it was enabled by default recently.

Reply:
> This module is used only in tests and in minimal ISO image.

No, it is also included for nixos-containers because otherwise the stupid ``environment.noXlibs`` option wouldn't mess with some software builds.

Reply:
I tested with this configuration:
```
  containers.test-man = {
    autoStart = true;
    privateNetwork = true;
    hostBridge = "br0";
    localAddress = "10.11.0.1/24";
    config = { pkgs, config, ... }: {
      system.extraDependencies = [ pkgs.hello ];
      services.openssh = {
        enable = true;
      };
    };
  };
```

Changes to `nixos/modules/profiles/minimal.nix` did not affect the container

Reply:
🤔 maybe I was doing something wonky but lxc container imports minimal profile which I was using for the container https://github.com/NixOS/nixpkgs/blob/master/nixos/modules/virtualisation/lxc-container.nix#L54

Reply:
Maybe make a separate profile for it?

Reply:
And now delete the minimal profile in lxc?

Reply:
@SuperSandro2000 add this patch?
```diff
diff --git a/nixos/doc/manual/from_md/release-notes/rl-2211.section.xml b/nixos/doc/manual/from_md/release-notes/rl-2211.section.xml
index 6a7032e1963..392f79cb51e 100644
--- a/nixos/doc/manual/from_md/release-notes/rl-2211.section.xml
+++ b/nixos/doc/manual/from_md/release-notes/rl-2211.section.xml
@@ -368,6 +368,11 @@
           <literal>nixos/modules/profiles/minimal.nix</literal> profile.
         </para>
       </listitem>
+      <listitem>
+        <para>
+          Now lxc containers don't use minimal profile.
+        </para>
+      </listitem>
       <listitem>
         <para>
           There is a new module for the <literal>xfconf</literal>
diff --git a/nixos/doc/manual/release-notes/rl-2211.section.md b/nixos/doc/manual/release-notes/rl-2211.section.md
index b56be16127b..817262affd8 100644
--- a/nixos/doc/manual/release-notes/rl-2211.section.md
+++ b/nixos/doc/manual/release-notes/rl-2211.section.md
@@ -134,6 +134,8 @@ Use `configure.packages` instead.

 - The minimal ISO image now use `nixos/modules/profiles/minimal.nix` profile.

+- Now lxc containers don't use minimal profile.
+
 - There is a new module for the `xfconf` program (the Xfce configuration storage system), which has a dbus service.

 <!-- To avoid merge conflicts, consider adding your item at an arbitrary place in the list instead. -->
diff --git a/nixos/modules/virtualisation/lxc-container.nix b/nixos/modules/virtualisation/lxc-container.nix
index d3a2e0ed151..b6b744dfbca 100644
--- a/nixos/modules/virtualisation/lxc-container.nix
+++ b/nixos/modules/virtualisation/lxc-container.nix
@@ -51,7 +51,6 @@ in
 {
   imports = [
     ../installer/cd-dvd/channel.nix
-    ../profiles/minimal.nix
     ../profiles/clone-config.nix
   ];

```

Reply:
Or this variant:
```diff
diff --git a/nixos/modules/virtualisation/lxc-container.nix b/nixos/modules/virtualisation/lxc-container.nix
index d3a2e0ed151..b6b4dcff1d0 100644
--- a/nixos/modules/virtualisation/lxc-container.nix
+++ b/nixos/modules/virtualisation/lxc-container.nix
@@ -55,6 +55,8 @@ in
     ../profiles/clone-config.nix
   ];

+  environment.noXlibs = lib.mkOverride 500 false;
+
   options = {
     virtualisation.lxc = {
       templates = mkOption {

```

Reply:
cc @SuperSandro2000

Reply:
@SuperSandro2000 Is there a consensus on this remark?

Reply:
noXlibs should not be enabled by default anywhere. It is an advanced setting that required multiple overrides to build on all my machines, which is not acceptable for normal users.

logrotation should not be disabled in lxc otherwise your disk can completely fill up which should be avoided with good defaults.